source/js/buyers-guide/components/creepometer/creepometer.jsx

```diff
@@ -4,12 +4,89 @@ export default class Creepometer extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {};
+    this.state = {
+      isHandleGrabbed: false,
+      handleOffset: 0
+    };
+
+    this.handleWidth = 70; // px
+    this.faceCount = 40; // Number of face frames
+    this.encodedStepCount = 100; // Upper range of values to be recorded
+    this.framePath = `/_images/buyers-guide/faces/`;
+
+    this.slideStart = this.slideStart.bind(this);
+    this.slideMove = this.slideMove.bind(this);
+    this.slideStop = this.slideStop.bind(this);
+
+    this.setSliderRef = element => {
+      this.sliderElement = element;
+    };
+  }
+
+  componentDidMount() {
+    // Slight delay because Firefox is too dang fast
+    setTimeout(() => {
+      // Set initial position
+      this.setState({
+        handleOffset: Math.floor(this.props.initialValue / this.encodedStepCount * this.sliderElement.scrollWidth),
+        encodedValue: this.props.initialValue
+      });
+    }, 100);
+  }
+
+  slideStart(e) {
+    if (e.nativeEvent.target.className === `handle`) {
+      this.setState({
+        isHandleGrabbed: true
+      });
+    }
+  }
+
+  slideMove(e) {
+    if (this.state.isHandleGrabbed) {
+      let clientX = e.nativeEvent.clientX;
+      let sliderLeftEdgeX = this.sliderElement.getBoundingClientRect().left;
+      let offset = Math.floor(clientX - sliderLeftEdgeX);
+
+      this.setState({
+        handleOffset: offset,
+        encodedValue: Math.floor(offset / this.sliderElement.scrollWidth * this.encodedStepCount)
+      });
+    }
+  }
+
+  slideStop() {
+    this.setState({
+      isHandleGrabbed: false
+    });
   }
 
   render() {
+    let handleX = this.state.handleOffset - this.handleWidth / 2;
+    let frameChoice = Math.floor(this.faceCount / 2);
+
+    // Don't let handle overflow slider's left side
+    handleX = handleX < 0 ? 0 : handleX;
+
+    if (this.sliderElement) {
+      // Don't let handle overflow slider's right side
+      if (handleX > this.sliderElement.scrollWidth - this.handleWidth) {
+        handleX = this.sliderElement.scrollWidth - this.handleWidth;
+      }
+
+      frameChoice = Math.floor(handleX / this.sliderElement.scrollWidth * this.faceCount) + 1;
+    }
+
+    let pxOffset = this.handleWidth * this.faceCount - this.handleWidth * frameChoice; // offset position for spritesheet
+
     return (
-      <div>Creepometer 😬</div>
+      <div class="creepometer">
+        <div className="slider" ref={this.setSliderRef} onMouseLeave={this.slideStop} onMouseMove={this.slideMove} onMouseDown={this.slideStart} onMouseUp={this.slideStop}>
```

Comment:
The onMouseLeave makes this feel weird. Do things get janky if you don't have it?

Reply:
That is, if you leave and then let up does it get "locked" as being held?

Reply:
Yeah, it will lock if you don't have it. You might try putting a larger wrapper around it with a onMouseLeave to make it a little less easy to release the handle.